src/RadixDlt.NetworkGateway.GatewayApi/Configuration/EndpointOptions.cs

```diff
@@ -106,6 +106,22 @@ public sealed class EndpointOptions
 
     [ConfigurationKeyName("TransactionStreamMaxFilterCount")]
     public int TransactionStreamMaxFilterCount { get; set; } = 10;
+
+    public int ResolvePageSize(int? requestPageSize) => ResolvePageSize(requestPageSize, DefaultPageSize, MaxPageSize);
```

Comment:
I don't think all these resolve methods belong in configuration class, it should be in some separate class that takes request value and configuration limit.

Reply:
I disagree. This is logic tightly bounded to the configuration itself. There's no point breaking the locality of the behavior for no good reason.